### PR TITLE
K8SPG-71 Version service PGO integration test

### PR DESCRIPTION
**Pull request by cap1984**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| scheduled-backup | failed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | passed |
| upgrade | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/159/commits/6ad8f2feb10460aea7a303338341f5e5a23734d2
image: `perconalab/percona-postgresql-operator:PR-159-6ad8f2fe-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-159-6ad8f2fe-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-159-6ad8f2fe-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-159-6ad8f2fe-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-159-6ad8f2fe-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-159-6ad8f2fe-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | passed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| scheduled-backup | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | passed |
| upgrade | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/159/commits/024793f8dafe4c61d95592f91a81678c91c10fbf
image: `perconalab/percona-postgresql-operator:PR-159-024793f8-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-159-024793f8-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-159-024793f8-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-159-024793f8-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-159-024793f8-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-159-024793f8-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| scheduled-backup | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | passed |
| upgrade | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/159/commits/ea2ea0782ffe0d97df1b351dfbe5a55872bbe190
image: `perconalab/percona-postgresql-operator:PR-159-ea2ea078-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-159-ea2ea078-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-159-ea2ea078-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-159-ea2ea078-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-159-ea2ea078-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-159-ea2ea078-pgo-deployer`

File: e2e-tests/version-service/run [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+	kubectl_bin exec $(kubectl_bin get pods --selector=name=pg-client -o 'jsonpath={.items[].metadata.name}') -- \
+		curl -s "${VS_ENDPOINT}/versions/v1/pg-operator/${version_to_upgrade}/${release_channel}" \
+		| jq -r '.versions[] | .matrix | .pgbackrest, .pgbackrestRepo, .pgbadger, .pgbouncer, .postgresql | .[].imagePath' \
+			>"${tmp_dir}/${version_to_upgrade}-origin-${release_channel}-images.txt"
+
+	diff -u "${tmp_dir}/${version_to_upgrade}-deployed-${release_channel}-images.txt" "${tmp_dir}/${version_to_upgrade}-origin-${release_channel}-images.txt"
+}
+
+main() {
```
Comment (hors) [inline e2e-tests/version-service/run:50]: Also we need to check the following case:
version of DB 13.5 but apply option was set `14-recommended` 

In this case cluster should not be updated.

desc 'PG cluster update with explicitly specified image inside version service'
desc 'PG cluster update with the latest image by version service'
desc 'PG cluster with version service offline'
desc 'Updating PG cluster with version service available but disabled'

**Review by hors — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | passed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| scheduled-backup | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | passed |
| upgrade | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/159/commits/4ca8d804706d9ab065591f17a6df585573510f80
image: `perconalab/percona-postgresql-operator:PR-159-4ca8d804-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-159-4ca8d804-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-159-4ca8d804-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-159-4ca8d804-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-159-4ca8d804-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-159-4ca8d804-pgo-deployer`

File: e2e-tests/version-service/run (hunk 1 above)
Comment (cap1984) [reply e2e-tests/version-service/run:48]: done

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| scheduled-backup | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | passed |
| upgrade | passed |
| smart-update | passed |
| version-service | failed |
| users | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/159/commits/af8c494d6bb6bb3e1f13857e6f09ecb5d610caeb
image: `perconalab/percona-postgresql-operator:PR-159-af8c494d-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-159-af8c494d-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-159-af8c494d-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-159-af8c494d-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-159-af8c494d-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-159-af8c494d-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| scheduled-backup | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | passed |
| upgrade | passed |
| smart-update | passed |
| version-service | failed |
| users | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/159/commits/f06e2c5e29ed2f8a69eea0cd8d51d2f23cda9e3d
image: `perconalab/percona-postgresql-operator:PR-159-f06e2c5e-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-159-f06e2c5e-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-159-f06e2c5e-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-159-f06e2c5e-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-159-f06e2c5e-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-159-f06e2c5e-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | passed |
| scaling | passed |
| scheduled-backup | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | passed |
| upgrade | passed |
| smart-update | passed |
| version-service | failed |
| data-migration-gcs | passed |
| users | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/159/commits/f06e2c5e29ed2f8a69eea0cd8d51d2f23cda9e3d
image: `perconalab/percona-postgresql-operator:PR-159-f06e2c5e-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-159-f06e2c5e-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-159-f06e2c5e-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-159-f06e2c5e-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-159-f06e2c5e-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-159-f06e2c5e-pgo-deployer`

**Review by hors — APPROVED** [review]

**Pull request merged**